packages/grafana-runtime/src/utils/userStorage.ts

```diff
@@ -0,0 +1,132 @@
+import { get } from 'lodash';
+import { lastValueFrom } from 'rxjs';
+
+import { config } from '../config';
+import { BackendSrvRequest, getBackendSrv } from '../services';
+
+const baseURL = `/apis/userstorage.grafana.app/v0alpha1/namespaces/${config.namespace}/user-storage`;
+
+interface RequestOptions extends BackendSrvRequest {
+  manageError?: (err: unknown) => { error: unknown };
+  showErrorAlert?: boolean;
+
+  // rtk codegen sets this
+  body?: BackendSrvRequest['data'];
+}
+
+export type UserStorageSpec = {
+  data: { [key: string]: string };
+};
+
+async function apiRequest<T>(requestOptions: RequestOptions) {
+  try {
+    const { data: responseData, ...meta } = await lastValueFrom(
+      getBackendSrv().fetch<T>({
+        ...requestOptions,
+        url: baseURL + requestOptions.url,
+        data: requestOptions.body,
+      })
+    );
+    return { data: responseData, meta };
+  } catch (error) {
+    return requestOptions.manageError ? requestOptions.manageError(error) : { error };
+  }
+}
+
+/**
+ * A class for interacting with the backend user storage.
+ */
+export class UserStorage {
+  private service: string;
+  private resourceName: string;
+  private userUID: string;
+  private canUseUserStorage: boolean;
+  private storageSpec: UserStorageSpec | null | undefined;
+
+  constructor(service: string) {
+    this.service = service;
+    this.userUID = config.bootData.user.uid === '' ? config.bootData.user.id.toString() : config.bootData.user.uid;
+    this.resourceName = `${service}:${this.userUID}`;
```

Comment:
Probably a silly question, but I guess the user UID postfix is validated on the backend, too?

Reply:
Correct, the backend checks that the user UID set in the name of the request and the one in the authentication info matches, if not, it returns a 403.